charts/agent-control/templates/installation-job.yaml

```diff
@@ -43,60 +65,77 @@ spec:
           {{- with .Values.installation.extraEnv }}
             {{- toYaml . | nindent 12 }}
           {{- end }}
+          volumeMounts:
+            {{- if .Values.agentControlCd.enabled }}
+            - name: flux-values
+              mountPath: /tmp/flux-values
+              readOnly: true
+            {{- if .Values.agentControlCd.repositoryCertificateSecretReferenceName }}
+            - name: flux-cert-secrets
+              mountPath: /etc/flux-cert
+              readOnly: true
+            {{- end }}
+            {{- end }}
+            {{- with .Values.installation.extraVolumeMounts }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          command: ["/bin/sh", "-c"]
           args:
             - |
+{{- if .Values.agentControlCd.enabled }}
               HELM_RELEASE_EXISTS=$(kubectl get HelmRelease agent-control-cd -n {{ .Release.Namespace }})
               if [ -z "$HELM_RELEASE_EXISTS" ]; then
                 echo "Adding agent-control-cd repository..."
                 helm repo add newrelic-agent-control-cd {{ .Values.agentControlCd.chartRepositoryUrl }}
-{{- /* Use TLS Options */ -}}
-{{- if .Values.agentControlCd.repositoryCertificateSecretReferenceName }} \
+              {{- /* Use TLS Options */ -}}
+              {{- if .Values.agentControlCd.repositoryCertificateSecretReferenceName }} \
                   --ca-file /etc/flux-cert/ca.crt \
                   --cert-file /etc/flux-cert/tls.crt \
                   --key-file /etc/flux-cert/tls.key
-{{- /* Use basic auth options */ -}}
-{{- else if .Values.agentControlCd.repositorySecretReferenceName }} \
+              {{- /* Use basic auth options */ -}}
+              {{- else if .Values.agentControlCd.repositorySecretReferenceName }} \
                   --username $HELM_REPO_USERNAME \
                   --password $HELM_REPO_PASSWORD
-{{- end }}
+              {{- end }}
                 echo "Installing agent-control-cd..."
                 helm upgrade --install agent-control-cd newrelic-agent-control-cd/{{ .Values.agentControlCd.chartName }} \
                   --version {{ .Values.agentControlCd.chartVersion | default .Chart.Annotations.agentControlCdVersion }} \
                   --namespace {{ .Release.Namespace }} \
                   -f /tmp/flux-values/agent-control-flux.yaml
               else
                 echo "agent-control-cd HelmRelease already exists, skipping installation."
-                exit 0
               fi
-          volumeMounts:
-            - name: flux-values
-              mountPath: /tmp/flux-values
-              readOnly: true
+{{- end }}
+{{- if .Values.agentControlDeployment.enabled }}
+           {{- if .Values.agentControlCd.enabled }}
+              echo "Creating agent-control-cd resources..."
```

Comment:
Mmmm, shouldn't me move that block on top? Otherwise, if a client enables `cd` and disables `deployment`, resources won't be created. I know it doesn't make sense for a client to do that, but I think it makes more sense.

Not needed to do now. It can be a sustain task.

Reply:
It is actually the expected behavior: the `agent-control-cd` resources (the CRs representing Flux's helm-release) are only useful if the Agent Control is installed, we don't need such resources if `agentControlDeployment` is disabled.